java/test/jmri/util/CheckHelpFilesForUTF8Test.java

```diff
@@ -0,0 +1,132 @@
+
+                var lines = Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8);
+                for (String s : lines) {
+                    s.codePoints().forEach((codePoint) -> {
+                        if (codePoint > 127) {
+                            numErrors++;
+                            foundChar.add(codePoint);
+                            String expected = convertChar.get(codePoint);
+                            log.error(
+                                    "Invalid character. Codepoint: {}, Character: {}, Replace with: {}, File: {}",
+                                    codePoint, new String(Character.toChars(codePoint)), expected, fileName);
+                        }
+                    });
+                }
+            }
+        }
+
+        Set<String> folders = Stream.of(path.toFile().listFiles())
+                  .filter(file -> file.isDirectory())
+                  .map(File::getName)
+                  .collect(Collectors.toSet());
+
+        for (String aFolder : folders) {
+            searchFolder(folder + aFolder + "/");
+        }
+
+    }
+
+    @Test
+    public void testGenerateSearchIndex() throws IOException {
+        // See: https://www.w3schools.com/charsets/ref_utf_punctuation.asp
+        convertChar.put(176, "&deg;");
+        convertChar.put(200, "&Egrave;");
+        convertChar.put(201, "&Eacute;");
+        convertChar.put(220, "&Uuml;");
+        convertChar.put(223, "&szlig;");
+        convertChar.put(224, "&agrave;");
+        convertChar.put(225, "&aacute;");
+        convertChar.put(226, "&acirc;");
+        convertChar.put(228, "&auml;");
+        convertChar.put(229, "&aring;");
+        convertChar.put(230, "&aelig;");
+        convertChar.put(231, "&ccedil;");
+        convertChar.put(232, "&egrave;");
+        convertChar.put(233, "&eacute;");
+        convertChar.put(234, "&ecirc;");
+        convertChar.put(237, "&iacute;");
+        convertChar.put(241, "&ntilde;");
+        convertChar.put(244, "&ocirc;");
+        convertChar.put(246, "&ouml;");
+        convertChar.put(248, "&oslash;");
+        convertChar.put(252, "&uuml;");
+        convertChar.put(253, "&yacute;");
+        convertChar.put(268, "&Ccaron;");
+        convertChar.put(283, "&ecaron;");
+        convertChar.put(339, "&oelig;");
+        convertChar.put(345, "&rcaron;");
+        convertChar.put(352, "&Scaron;");
+        convertChar.put(381, "&Zcaron;");
+        convertChar.put(8209, "&#8209;");
+        convertChar.put(8211, "&ndash;");
+        convertChar.put(8212, "&mdash;");
+        convertChar.put(8216, "&lsquo;");
+        convertChar.put(8217, "&rsquo;");
+        convertChar.put(8220, "&ldquo;");
+        convertChar.put(8221, "&rdquo;");
+        convertChar.put(8226, "&bull;");
+        convertChar.put(8230, "&hellip;");
+        convertChar.put(8250, "&rsaquo;");
+        convertChar.put(8482, "&trade;");
+        convertChar.put(8594, "&rarr;");
```

Comment:
8658 `&rArr;` is missing.

Reply:
I have added `&rArr;` now. I didn't added it before because I made the list from the characters that was found and `&rArr;` was not one of those. I created this list to make it easier to resolve this issue.

Reply:
You would not find the `&rArr;` errors, along with `&reg;` and `&copy;`, since those were all fixed in PR #10818.

I guess that means you need to add reg and copy to the list.

Reply:
I have added `&reg;` and `&copy;` to the list now. But there is too many UTF-8 characters so it's impossible for the list to be complete.

See this page and its siblings for a good reference:
https://www.w3schools.com/charsets/ref_utf_punctuation.asp

Reply:
The list does not need to be complete, it just needs to cover the entities that we commonly use.

Reply:
I'm happy to add more items if desired.